lhotse/features/mixer.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+
+import numpy as np
+
+from lhotse.features.base import FeatureExtractor
+from lhotse.utils import Seconds, Decibels
+
+
+class FeatureMixer:
+    """
+    Utility class to mix multiple feature matrices into a single one.
+    It pads the signals with low energy values to account for differing lengths and offsets.
+    It relies on the ``FeatureExtractor`` to have defined ``mix`` and ``compute_energy`` methods,
+    so that the ``FeatureMixer`` knows how to scale and add two feature matrices together.
+    """
+
+    def __init__(
+            self,
+            feature_extractor: FeatureExtractor,
+            base_feats: np.ndarray,
+            frame_shift: Seconds,
+            log_energy_floor: float = -1000.0
+    ):
+        """
+        :param feature_extractor: The ``FeatureExtractor`` that specifies how to mix the features.
+        :param base_feats: The features used to initialize the ``FeatureMixer`` are a point of reference
+            in terms of energy and offset for all features mixed into them.
+        :param frame_shift: Required to correctly compute offset and padding during the mix.
+        :param log_energy_floor: The value used to pad the shorter features during the mix.
+        """
+        self.feature_extractor = feature_extractor
+        # The mixing output will be available in self.mixed_feats
+        self.mixed_feats = base_feats
+        # Keep a pre-computed energy value of the features that we initialize the Mixer with;
+        # it is required to compute gain ratios that satisfy SNR during the mix.
+        self.frame_shift = frame_shift
+        self.reference_energy = feature_extractor.compute_energy(base_feats)
```

Comment:
is there a requirement that energy never be zero?  Please clarify where compute_energy is documented if so!